app/lib/three_scale/socket.rb

```diff
@@ -0,0 +1,13 @@
+module ThreeScale
+  module Socket
+    module ClassMethods
+      def accepts_connections?(host, port)
+        ::Socket.tcp(host, port, connect_timeout: 1) { true }
+      rescue
+        false
+      end
+    end
+
+    extend ClassMethods
```

Comment:
This design is a bit tricky, are you trying to avoid extending `ActiveSupport::Concern`?

Reply:
My bad!, I assumed that you couldn't use `ClassMethods` unless you include `ActiveSupport::Concern`.